ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/HeadObserver.scala

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.infrastructure
+
+import io.grpc.Context
+import io.grpc.stub.StreamObserver
+
+import scala.concurrent.{Future, Promise}
+
+object HeadObserver {
```

Comment:
How is this different than `FirstElementObserver` and can we get rid of one of them?

Reply:
This cancels the call after we received the first item, so I believe this should not leak open streams. I preferred to duplicate it to not introduce a dependency between the test tool and the integration tests. We can put it in a shared library and get rid of the `FirstElementObserver`. Sounds good?

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2282/commits/55bd945701279f2b4c8a83f6adf3b57a7f67d983

Renamed to `SingleItemObserver`, made it either get the very first item of the stream or the first that satisfies a predicate, moved it to `//ledger/test-common`, deleted `FirstElementObserver`, replaced all usages. This should be it. :sweat_smile: